src/Microsoft.ML.Data/Transforms/RowShufflingTransformer.cs

```diff
@@ -559,28 +559,28 @@ protected override void Dispose(bool disposing)
 
                 if (disposing)
                 {
-                    _toProduce.Complete();
+                    _toProduceChannel.Writer.Complete();
                     _producerTask.Wait();
 
                     // Complete the consumer after the producerTask has finished, since producerTask could
                     // have posted more items to _toConsume.
-                    _toConsume.Complete();
+                    _toConsumeChannel.Writer.Complete();
 
                     // Drain both BufferBlocks - this prevents what appears to be memory leaks when using the VS Debugger
                     // because if a BufferBlock still contains items, its underlying Tasks are not getting completed.
                     // See https://github.com/dotnet/corefx/issues/30582 for the VS Debugger issue.
                     // See also https://github.com/dotnet/machinelearning/issues/4399
-                    _toProduce.TryReceiveAll(out _);
-                    _toConsume.TryReceiveAll(out _);
+                    _toProduceChannel.Reader.ReadAsync();
```

Comment:
We should test that this is still necessary. Read the linked issues for more information on why this was added. We might not need to drain these anymore with the move to Channel.

Reply:
If we do still need to drain these channels, this is wrong:

1. You need to synchronously read the item (you are doing `ReadAsync()` here and ignoring the result)
2. You need to read all of the messages, not just the first one

Reply:
I did a very rudimentary test using a test console app that was in [#26565](https://github.com/dotnet/runtime/issues/26565) from @alandoherty. I saw the issue with `BufferBlock` but when changing to `Channels` I didn't see the issue. Even with and without completing the writer, the memory stayed consistent.

This may be safe to remove, but I'll double check by running the benchmark tests in a before and after to see if there are any big jumps in the numbers.

Reply:
@eerhardt Ran a couple of benchmarks and below are some results. It did complain that I didn't have it compiled in Release mode even though I ran the build script with it and built in Release in Visual Studio so these results may be under Debug mode.

They look very similar with and without having to flush out the Channels. I'm thinking this may be safe to remove but curious about your thoughts.

Without update:
|            Method |   Mean |  Error | StdDev | Extra Metric |       Gen 0 |      Gen 1 |     Gen 2 | Allocated |
|------------------- |--------:|--------:|--------:|-------------:|-------------:|------------:|-----------:|----------:|
| TrainFeaturizeText | 23.30 s | 23.88 s | 1.309 s |           - | 1997000.0000 | 781000.0000 | 40000.0000 | 320.3 MB |
 
|         Method |   Mean |   Error | StdDev | Extra Metric |     Gen 0 |     Gen 1 |     Gen 2 | Allocated |
|---------------- |--------:|---------:|--------:|-------------:|-----------:|-----------:|-----------:|----------:|
| TrainResnetV250 | 67.38 s | 283.41 s | 15.53 s |           - | 75000.0000 | 34000.0000 | 11000.0000 |  2.01 GB |
 
With update:
|            Method |   Mean |  Error | StdDev | Extra Metric |       Gen 0 |      Gen 1 |     Gen 2 | Allocated |
|------------------- |--------:|--------:|--------:|-------------:|-------------:|------------:|-----------:|----------:|
| TrainFeaturizeText | 19.77 s | 29.54 s | 1.619 s |           - | 1978000.0000 | 771000.0000 | 34000.0000 | 320.3 MB |
 
|         Method |   Mean |   Error | StdDev | Extra Metric |     Gen 0 |     Gen 1 |     Gen 2 | Allocated |
|---------------- |--------:|---------:|--------:|-------------:|-----------:|-----------:|-----------:|----------:|
| TrainResnetV250 | 54.75 s | 111.72 s | 6.124 s |           - | 76000.0000 | 34000.0000 | 11000.0000 |  2.01 GB |

Reply:
> I saw the issue with BufferBlock but when changing to Channels I didn't see the issue. Even with and without completing the writer, the memory stayed consistent.

Then I think we are OK to remove this "draining" code altogether.